app/models/post.rb

```diff
@@ -0,0 +1,15 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :comments, dependent: :destroy
+  has_many :likes, dependent: :destroy
+
+  after_create :update_posts_count_on_create
+
+  def five_most_recent_comments
+    comments.order(created_at: :desc).limit(5)
+  end
+
+  def update_posts_count_on_create
+    author.increment!(:posts_counter)
+  end
```

Comment:
- [ OPTIONAL ]  Hey there! 🌟 Your code is fantastic! To take it up a notch, I'd suggest marking the method responsible for updating the post counter as private. 🛡️ Why? Because you've already set up a Rails active callback that fires when a new post is added. This callback then calls the method to update the post counter. 🔄 By marking it private, you're signaling that it should only be accessed within the class, keeping things neat and secure. 👌 Great job! 🚀